### DISCUSSION TITLE
Request: Control volume of a specific program with hotkey

### DISCUSSION THREAD

**Discussion by DracoClaw** [File-New-Project/EarTrumpet#684] [Completed ideas]
So i just found this and im in love, my only qusetion is if i can control the volume of a specific program with a key or even better the volume scroll on my keyboard.

**Comment by HoZe-Lukas** [comment]
This feature would be even better if you could use a midi board with faders/nobs (like the behringer x-touch mini) to control the different programs by turning the knobs. or at least have a interface so we can create plugins to do that

**Comment by m0ck0** [comment]
pleaaaaaaseee, this would make the app just perfect

**Comment by Packetdancer** [comment]
FWIW, I have a personal fork of EarTrumpet that lets me do basically this, including using modifiers with the keyboard volume control; you can target either a device or an application ON a specific device. The only aspect of this that it doesn't handle right now would be adjusting the volume of an application _regardless_ of what audio device it was on.

https://raw.githubusercontent.com/Packetdancer/EarTrumpet/master/Graphics/hero5.png will show what I mean. 

Mostly I just did it because I wanted this functionality myself; I can adjust the volume of, say, Discord and a game that I'm streaming individually this way. But I'd be happy to make a pull request to contribute it back, if others think it would be useful. Though admittedly the architecture to preserve the volume target functionality even when an application has been exited and restarted got a little more tangled than I originally expected; it could probably do with some cleanup.

**Comment by DracoClaw** [comment]
Would i be able to use the volume scroll wheel on my G910?

**Comment by Packetdancer** [comment]
I actually wrote the fork with a Logitech G910 Orion Spark as my keyboard, though I replaced it with a Corsair K95 last week. (Yay new keyboard!) So I think it's safe to say that while the forked version of EarTrumpet _should_ work with every keyboard's media controls, I can _guarantee_ it works with only two, and the G910 is one of them. ;)

**Comment by DracoClaw** [comment]
Oh! That's amazing, Id love to try that out!

**Comment by DracoClaw** [comment]
@Packetdancer I never heard back from you, but how could i go about using your version?

**Comment by DracoClaw** [comment]
Update: Ive been using their fork for a while and its very good

**Comment by Diagonet** [comment]
I would kill for this feature, so many times I've had to alt+tab just to mute twitch/youtube I've lost count

**Reply by riverar** [reply]
You're in luck because it's done. As soon as I merge the code into dev (tonight), you can install dev builds of EarTrumpet and use it.

**Reply by Diagonet** [reply]
Okay, that was fast! haha  
Thanks for all the work you do!

**Reply by riverar** [reply]
It's live now. 🎉 Open Settings > Actions and create a new EarTrumpet Action. (e.g. When [a hotkey is typed] Do mute [application]. Please file any bugs, UI issues, etc. you come across!

**Reply by Diagonet** [reply]
Oh yeah, downloaded it and its great! Only issue so far is not being able to keybind Alt+F9

**Reply by DracoClaw** [reply]
So IDK if its something im doing, but i cant seem to set for example: Ctrl + Volume = Adjust Spotify

**Comment by DracoClaw** [comment]
Hey there! Im on the latest build 2.1.9.0 and im not seeing these features. Any help would be appreciated

**Reply by DracoClaw** [reply]
Found the dev build 2.1.9.18 and the feature, thanks

**Comment by riverar** [comment]
This feature is available in EarTrumpet dev builds. Open Settings > Actions and create a new EarTrumpet Action. (e.g. When [a hotkey is typed] Do mute [application]. Please file any bugs, UI issues, etc. you come across.

Moving to completed ideas.

**Reply by cmcpasserby** [reply]
Will this be available in non dev builds, am running 2.2.0.0 and do not see it. If its not, is there a any work i could do to speed it along?

**Reply by riverar** [reply]
@cmcpasserby The dev builds aren't far off from main, so I'd just switch to that. Progress is slow as I'm the sole maintainer right now and Windows app development ecosystem is in flux.

**Reply by cmcpasserby** [reply]
thanks also understandable MS is always a bit of a moving target. So you feel the dev branch is stable enough for daily use then?

**Reply by riverar** [reply]
@cmcpasserby I do. And you'll get bug fixes much quicker, which is a plus.

**Comment by Diagonet** [comment]
Has this been brought to the main branch? Can I stop using the dev branch now?

**Comment by Rajdave69** [comment]
It's been a year since the last reply. Yet this feature is still not added into the main branch, 4 years after implementation.

Bruh

**Reply by riverar** [reply]
The `dev` branch isn't far off from `main`, does it really matter?
